support/android/apk/servoapp/src/main/java/org/mozilla/servo/MainActivity.java

```diff
@@ -162,6 +163,14 @@ public void onHistoryChanged(boolean canGoBack, boolean canGoForward) {
         mCanGoBack = canGoBack;
     }
 
+    public void onAllowNavigation(String url) {
```

Comment:
I think your exception come from somewhere in here. So you can surround this code with a try/catch to see what crashes here.